### PR TITLE
refactor(action): tailwindify

### PR DESCRIPTION
**Pull request by jrowlingson**
Supports #1500

This changeset refactors the existing styles to utilize the Tailwind config. Sass usage has mostly been removed with the exception of `%component-host` which can be addressed later.

- [x] Merge https://github.com/Esri/calcite-components/pull/1520

### CODE REVIEW

File: src/components/calcite-action/calcite-action.scss [hunk 1]
```diff
@@ -1,246 +1,230 @@
...
 }
 
 .slot-container {
-  display: flex;
+  @apply flex;
 }
 
 .slot-container--hidden {
-  display: none;
+  @apply hidden;
```
Comment (eriklharper) [inline src/components/calcite-action/calcite-action.scss:91]: What would be the tailwind rule for `visibility: hidden` if `hidden` corresponds to `display: none`?  Asking for a friend... ;)

File: src/components/calcite-action/calcite-action.scss [hunk 2]
```diff
@@ -1,246 +1,230 @@
...
-    align-items: center;
-    display: flex;
-    pointer-events: none;
+    @apply flex
+      items-center
+      justify-center
+      m-0
+      pointer-events-none;
+    min-width: theme("spacing.4");
+    min-height: theme("spacing.4");
```
Comment (eriklharper) [inline src/components/calcite-action/calcite-action.scss:45]: Also here, `items-center` is vague.  Does it apply to `align-items` or `justify-items`?

File: src/components/calcite-action/calcite-action.scss (hunk 1 above)
Comment (jrowlingson) [reply src/components/calcite-action/calcite-action.scss:91]: `invisible` :exploding_head:

File: src/components/calcite-action/calcite-action.scss (hunk 2 above)
Comment (jrowlingson) [reply src/components/calcite-action/calcite-action.scss:45]: Ah, yeah, the learning curve on the name choices is a bit unfortunate. The brevity is coming from the inline use
case. In practice, it becomes second nature after some use.


| Property | Utility |
| -------- | ------- |
| align-items | items-* |
| align-content | content-* |
| align-self | self-* |
| justify-content | justify-* |

Where `*` is the value. In the case of a hypenated value, the last word is used e.g.:

`justify-content: space-evenly;` -> `justify-evenly`

Comment (eriklharper) [reply src/components/calcite-action/calcite-action.scss:45]: Cool, thanks for the info on that, I'll be sure to reference back to this!

**Review by eriklharper — APPROVED** [review]

**Pull request merged**